app/controllers/publish/publish_controller.rb

```diff
@@ -9,6 +9,9 @@ class PublishController < ApplicationController
     before_action :check_interrupt_redirects
     before_action :clear_previous_cycle_year_in_session, unless: -> { FeatureService.enabled?('rollover.can_edit_current_and_next_cycles') }
 
+    # Protect every action of a provider
+    before_action { authorize provider, :show? if provider }
```

Comment:
Do we need this comment?

Also, it might be just me, but it might be easier to read either:

1. Create an intention-revealing method name for this:

```suggestion
    before_action :authorize_provider_sub_pages

    def authorize_provider_sub_pages
      authorize provider, :show? if provider
    end
```

3. Pull down the code into a multiline block:

```suggestion
    before_action do
      authorize provider, :show? if provider
    end
```

Reply:
I know I'm not as averse to code comments as other people are. I can remove the comment.

I wonder if there is more work to be done here if I need to add the `if provider` exists.